lib/cisco_node_utils/bgp.rb

```diff
@@ -152,6 +158,17 @@ def bestpath_med_non_deterministic
       config_get('bgp', 'bestpath_med_non_deterministic', @get_args)
     end
 
+    # nsr Setter
+    def nsr=(enable)
+      if platform == :nexus || @vrf != 'default'
+        fail Cisco::UnsupportedError.new('bgp', 'nsr', 'set', 'nsr is not ' \
```

Comment:
Oh, good point.  Then the message should be simply:

"nsr is not configurable ' \
'on a per-VRF basis"

Or will that be annoying to the end-user?  Imagine on nexus, they first try to configure nsr on a non-default vrf and get the "nsr is not configurable on a per-VRF basis" message.  They say "oh, I'll just set it for global BGP" and then get the normal unsupported error.  "Why didn't you just tell me that to begin with?"

This is really nitpicking, so it's fine whichever way you want to do it.